core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -564,21 +563,19 @@ public ServerBuilder serviceLoggerPrefix(String serviceLoggerPrefix) {
 
     /**
      * Sets the format of this {@link Server}'s access log. The specified {@code accessLogFormat} would be
-     * parsed by {@link AccessLogWriters#custom(String)}.
+     * parsed by {@link AccessLogWriter#custom(String)}.
      */
     public ServerBuilder accessLogFormat(String accessLogFormat) {
-        accessLogWriter = AccessLogWriters.custom(requireNonNull(accessLogFormat, "accessLogFormat"));
-        return this;
+        return accessLogWriter(AccessLogWriter.custom(requireNonNull(accessLogFormat, "accessLogFormat")),
+                               true);
     }
 
     /**
-     * Sets an access log writer of this {@link Server}. {@link AccessLogWriters#disabled()} is used by default.
-     *
-     * @see AccessLogWriters to find pre-defined access log writers.
+     * Sets an access log writer of this {@link Server}. {@link AccessLogWriter#disabled()} is used by default.
      */
-    @SuppressWarnings("unchecked")
-    public ServerBuilder accessLogWriter(Consumer<? super RequestLog> accessLogWriter) {
-        this.accessLogWriter = (Consumer<RequestLog>) requireNonNull(accessLogWriter, "accessLogWriter");
+    public ServerBuilder accessLogWriter(AccessLogWriter accessLogWriter, boolean shutdownOnStop) {
```

Comment:
Ah, I see. I thought that the AccessLogWriter can control the property by implementing its `shutdown` method. But I realized that it might also be controlled by a user who builds the server...

Reply:
How about adding `@param` comment about the `shutdownOnStop` parameter?

Reply:
Added.